### PR TITLE
USHIFT-2118: Add 2 RF test for Cluster ID feature

### PR DESCRIPTION
**Pull request by agullon**
**Which issue(s) this PR addresses**:

Closes [USHIFT-2118 ](https://issues.redhat.com/browse/USHIFT-2118)

### CODE REVIEW

File: test/suites/osconfig/clusterid.robot [hunk 1]
```diff
@@ -83,10 +102,79 @@ Get MicroShift Cluster ID From Namespace
...
     ...    sed -i '$ s/$/123/' ${CLUSTERID_FILE}
-    ...    sudo=True    return_rc=True    return_stdout=True    return_stderr=True
+    ...    sudo=True    return_rc=True    return_stdout=False
+
+    Should Be Equal As Integers    0    ${rc}
+
+Remove Cluster ID File
+    [Documentation]    Append invalid characters to the cluster ID file.
+    ${rc}=    Execute Command
+    ...    rm -rf ${CLUSTERID_FILE}
```
Comment (ggiguash) [inline test/suites/osconfig/clusterid.robot:154]: We should only use `-r` option when deleting directories.
```suggestion
    ...    rm -f ${CLUSTERID_FILE}
```

File: test/suites/osconfig/clusterid.robot [hunk 2]
```diff
@@ -83,10 +102,79 @@ Get MicroShift Cluster ID From Namespace
...
 
+Extract Sos Report
+    [Documentation]    Extract Sos Report from the tar file
+    [Arguments]    ${sos_report_tarfile}
+
+    ${sos_report_dir}    ${rc}=    Execute Command
+    ...    dirname ${sos_report_tarfile}
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Log    ${sos_report_dir}
+    Should Be Equal As Integers    0    ${rc}
```
Comment (ggiguash) [inline test/suites/osconfig/clusterid.robot:113]: Please, use RF-native functionality for path parsing, i.e. `Operating System.Get Parent Directory`

File: test/suites/osconfig/clusterid.robot [hunk 3]
```diff
@@ -83,10 +102,79 @@ Get MicroShift Cluster ID From Namespace
...
+
+    ${rc}=    Execute Command
+    ...    sos report --batch --all-logs --tmp-dir ${sos_report_dir} -p microshift -o logs
+    ...    sudo=True    return_rc=True    return_stdout=False
+    Should Be Equal As Integers    0    ${rc}
+
+    ${sos_report_tarfile}    ${rc}=    Execute Command
+    ...    find ${sos_report_dir} -type f -name "sosreport-*.tar.xz"
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Should Be Equal As Integers    0    ${rc}
```
Comment (ggiguash) [inline test/suites/osconfig/clusterid.robot:178]: We can simplify this code by using `Command Should Work` from `microshift-host.resource`

File: test/suites/osconfig/clusterid.robot [hunk 4]
```diff
@@ -83,10 +102,79 @@ Get MicroShift Cluster ID From Namespace
...
+
+    ${rc}=    Execute Command
+    ...    tar xf ${sos_report_tarfile} -C ${sos_report_dir}
+    ...    sudo=True    return_rc=True    return_stdout=False
+    Should Be Equal As Integers    0    ${rc}
+
+    ${sos_report_untared}    ${rc}=    Execute Command
+    ...    find ${sos_report_dir} -type d -name "sosreport-*"
+    ...    sudo=True    return_rc=True    return_stdout=True
+    Should Be Equal As Integers    0    ${rc}
```
Comment (ggiguash) [inline test/suites/osconfig/clusterid.robot:123]: We can simplify this code by using `Command Should Work` from `microshift-host.resource`

**Review by ggiguash — CHANGES_REQUESTED** [review]

File: test/suites/osconfig/clusterid.robot [hunk 5]
```diff
@@ -83,10 +102,79 @@ Get MicroShift Cluster ID From Namespace
...
+Get MicroShift Cluster ID From Sos Report
+    [Documentation]    Read and return the Cluster ID from the kube-system namespace yaml description in the Sos Report.
+    [Arguments]    ${sos_report_tarfile}
+
+    ${sos_report_untared}=    Extract Sos Report    ${sos_report_tarfile}
+
+    ${namespace_yaml_path}=    Set Variable
+    ...    ${sos_report_untared}/sos_commands/microshift/namespaces/${CLUSTERID_NS}/${CLUSTERID_NS}.yaml
+    ${clusterid}    ${rc}=    Execute Command
+    ...    cat ${namespace_yaml_path} | sed -n 's/\\s\\suid:\\s//p'
```
Comment (ggiguash) [inline test/suites/osconfig/clusterid.robot:136]: Please, use `Yaml Parse` function as in here for example:
https://github.com/openshift/microshift/blob/main/test/resources/microshift-process.resource#L23

**Review by ggiguash — CHANGES_REQUESTED** [review]

File: test/suites/osconfig/clusterid.robot (hunk 5 above)
Comment (agullon) [reply test/suites/osconfig/clusterid.robot]: thanks, added in the last commit

File: test/suites/osconfig/clusterid.robot (hunk 4 above)
Comment (agullon) [reply test/suites/osconfig/clusterid.robot]: thanks, replaced in all `Execute Command` keywords

File: test/suites/osconfig/clusterid.robot (hunk 3 above)
Comment (agullon) [reply test/suites/osconfig/clusterid.robot]: thanks, I agree, I didn't know about this keyword

File: test/suites/osconfig/clusterid.robot (hunk 1 above)
Comment (agullon) [reply test/suites/osconfig/clusterid.robot]: changed

File: test/suites/osconfig/clusterid.robot (hunk 2 above)
Comment (ggiguash) [reply test/suites/osconfig/clusterid.robot]: Sorry, we should use `Split Path` keyword from https://robotframework.org/robotframework/latest/libraries/OperatingSystem.html#Split%20Path

Comment (agullon) [reply test/suites/osconfig/clusterid.robot]: thanks, added `Split Path` keyword

**Pull request merged**